packages/config/package.json

```diff
@@ -0,0 +1,32 @@
+{
+  "name": "@packages/config",
+  "version": "0.0.0-development",
+  "description": "Config contains the configuration types and validation function used in the cypress electron application.",
+  "private": true,
+  "main": "lib/index.js",
+  "scripts": {
+    "build": "tsc --project .",
```

Comment:
I messed with this a bit and this was worked for both local and prod tests/builds since typescript is compile `src` to `lib` when build is ran. Can you explain the piggy-backing concept a bit more? Does this work because the typescript files in `packages/launcher` & `packages/proxy` are being compiled to the same directory as the `.ts` files are located?

Reply:
> Does this work because the typescript files in packages/launcher & packages/proxy are being compiled to the same directory as the .ts files are located?

Yeah, that's right. Is there any reason we can't do that here? I think we should because otherwise if you change a src file in `packages/config` in the course of development, it will be out-of-date unless you re-run the build script (or we'd need to set up a watch to keep the build up-to-date).

Reply:
We could, but it was kind of confusing to have compiled & non-compiled code within the same directory. Several of the packages weren't cleaning their compiled files up correctly and I had to go through & delete them because some have .ts and .js files intermixed before the compiler is ran.

Reply:
Since it will only build for production, which only happens in CI, there wouldn't be compiled files alongside non-compiled files locally while developing. In which packages did you see compiled files? That should only happen if `build-prod` is run.

Reply:
I had errors in CI with build-prod so I was running it locally to debug and noticed the behavior. I'm not totally stuck on this change, just felt it was clearer IMO. I removed it here: [`8bdfd57` (#18589)](https://github.com/cypress-io/cypress/pull/18589/commits/8bdfd57be4cc3115b94c4013a81d8b629f81d318)